pkg/plugin/state.go

```diff
@@ -695,6 +695,33 @@ func (e *AutoscaleEnforcer) handleUpdatedScalingBounds(vm *api.VmInfo, unqualifi
 	klog.Infof(fmtString, podName, pod.node.name, cpuVerdict, memVerdict)
 }
 
+func (e *AutoscaleEnforcer) handleNonAutoscalingUsageChange(vm *api.VmInfo, name util.NamespacedName) {
+	e.state.lock.Lock()
+	defer e.state.lock.Unlock()
+
+	klog.Info("Handling updated usage information for non-autoscaling VM pod %v", name)
+
+	pod, ok := e.state.podMap[name]
+	if !ok {
+		klog.Errorf("[autoscale-enforcer] cannot find non-autoscaling VM pod %v to update usage", name)
+		return
+	}
+	node := pod.node
+
+	newMem, newCPU := vm.Using().Mem, vm.Using().VCPU
+	oldMem, oldCPU := pod.memSlots.Reserved, pod.vCPU.Reserved
+
+	node.memSlots.Reserved, node.vCPU.Reserved = newMem, newCPU
```

Comment:
This line is incorrect — `node.{memSlots,vCPU}.Reserved` is the sum of all pods on the node; we need to update the node's resources based on the change in the pod (instead of e.g. setting it _equal_ to the pod's)

---

as an aside: This kind of error is easy to make. While it's not necessary here, this is partly why I've tended to make the handler functions generic over the resource. I find less repetition makes it easier to parse what's actually going on (of course, there's two sides to every coin — wanted to share some motivation behind why things are the way they are)

Reply:
Yup. For Kubernetes in general, Nodes have multiple Pods; Pods each have ≥ 1 Container (defined at creation of the Pod). For us, we only have one Container per-Pod, and it's running the VM.